lib/msal-browser/src/operatingcontext/BaseOperatingContext.ts

```diff
@@ -24,15 +29,67 @@ export abstract class BaseOperatingContext {
     protected available: boolean;
     protected browserEnvironment: boolean;
 
+    protected static loggerCallback(level: LogLevel, message: string): void {
+        switch (level) {
+            case LogLevel.Error:
+                // eslint-disable-next-line no-console
+                console.error(message);
+                return;
+            case LogLevel.Info:
+                // eslint-disable-next-line no-console
+                console.info(message);
+                return;
+            case LogLevel.Verbose:
+                // eslint-disable-next-line no-console
+                console.debug(message);
+                return;
+            case LogLevel.Warning:
+                // eslint-disable-next-line no-console
+                console.warn(message);
+                return;
+            default:
+                // eslint-disable-next-line no-console
+                console.log(message);
+                return;
+        }
+    }
+
     constructor(config: Configuration) {
         /*
          * If loaded in an environment where window is not available,
          * set internal flag to false so that further requests fail.
          * This is to support server-side rendering environments.
          */
         this.browserEnvironment = typeof window !== "undefined";
-
         this.config = buildConfiguration(config, this.browserEnvironment);
+
+        let sessionStorage: Storage | undefined;
+        try {
+            sessionStorage = window[BrowserCacheLocation.SessionStorage];
+            // Mute errors if it's a non-browser environment or cookies are blocked.
+        } catch (e) {}
+
+        const logLevelKey = sessionStorage?.getItem(LOG_LEVEL_CACHE_KEY);
+        const piiLoggingEnabled =
+            sessionStorage?.getItem(LOG_PII_CACHE_KEY)?.toLowerCase() ===
+            "true";
+        const loggerOptions = { ...this.config.system.loggerOptions };
+
+        if (logLevelKey || piiLoggingEnabled) {
+            const logLevel =
+                logLevelKey && Object.keys(LogLevel).includes(logLevelKey)
+                    ? LogLevel[logLevelKey]
+                    : undefined;
+            if (logLevel) {
+                loggerOptions.loggerCallback =
+                    BaseOperatingContext.loggerCallback;
+                loggerOptions.logLevel = logLevel;
+            }
+            if (piiLoggingEnabled) {
+                loggerOptions.piiLoggingEnabled = piiLoggingEnabled;
+            }
+        }
```

Comment:
Outer `if` is redundant
```suggestion
            const logLevel =
                logLevelKey && Object.keys(LogLevel).includes(logLevelKey)
                    ? LogLevel[logLevelKey]
                    : undefined;
            if (logLevel) {
                loggerOptions.loggerCallback =
                    BaseOperatingContext.loggerCallback;
                loggerOptions.logLevel = logLevel;
            }
            if (piiLoggingEnabled) {
                loggerOptions.piiLoggingEnabled = piiLoggingEnabled;
            }
```

Reply:
Fixed.